planet/auth.py

```diff
@@ -0,0 +1,285 @@
+# Copyright 2020 Planet Labs, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''Manage authentication with Planet APIs'''
+from __future__ import annotations  # https://stackoverflow.com/a/33533514
+import abc
+import base64
+import json
+import logging
+import os
+
+import httpx
+
+from . import constants
+from .api import http, models
+
+AUTH_URL = constants.PLANET_BASE_URL + 'v0/auth/'
+
+ENV_API_KEY = 'PL_API_KEY'
+
+SECRET_FILE_PATH = os.path.join(os.path.expanduser('~'), '.planet.json')
```

Comment:
Just commenting to say this made me check the docs for `expanduser()` and TIL it works for Windows envs too. Nice!

Reply:
thanks! That's right outta V1, so was also a learning exercise for me!